### PR TITLE
Expose host property from prometheus/common

### PR DESCRIPTION
**Pull request by jkroepke**
#### PR Description

In https://github.com/prometheus/common/pull/549 I add the possibility to define a custom HTTP header. I would like to use that feature in Grafana Alloy as well.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/reference/components/discovery.azure.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Name                     | Type                | Description
 `proxy_from_environment` | `bool`              | Use the proxy URL indicated by environment variables.                                            | `false`              | no
 `proxy_connect_header`   | `map(list(secret))` | Specifies headers to send to proxies during CONNECT requests.                                    |                      | no
 `follow_redirects`       | `bool`              | Whether redirects returned by the server should be followed.                                     | `true`               | no
+`host`                   | `string`            | Override HTTP hostname                                                                           |         | no
```
Comment (clayton-cornell) [inline docs/sources/reference/components/discovery.azure.md:35]:
```suggestion
`host`                   | `string`            | Override HTTP hostname.                                                                           |         | no
```

For consistency, we should add a full stop (period) at the end of this description.  This should be cascaded to to all instances of `Override HTTP hostname.`

**Comment by clayton-cornell** [comment]
Over to @grafana/grafana-alloy-maintainers for code review

**Comment by clayton-cornell** [comment]
Ping for review? @grafana/grafana-agent-maintainers

File: internal/component/common/config/types.go [hunk 2]
```diff
@@ -92,6 +93,7 @@ func (h *HTTPClientConfig) Convert() *config.HTTPClientConfig {
 		FollowRedirects: h.FollowRedirects,
 		EnableHTTP2:     h.EnableHTTP2,
 		ProxyConfig:     h.ProxyConfig.Convert(),
+		Host:            h.Host,
```
Comment (ptodev) [inline internal/component/common/config/types.go:96]: @jkroepke It looks like `Host` [got removed](https://github.com/prometheus/common/pull/597) from the upstream repo.

Comment (jkroepke) [reply internal/component/common/config/types.go:96]: yes,  see https://github.com/prometheus/common/pull/645

**Pull request closed without merging**